server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -367,6 +367,14 @@ public void forceTermsAggsToExcludeDeletedDocs() {
                     TermsAggregationBuilder termsBuilder = (TermsAggregationBuilder) current;
                     if (termsBuilder.minDocCount() == 0) {
                         termsBuilder.excludeDeletedDocs(true);
+                        // 7.17.x serialization compatibility
+                        // due to a long-standing issue with transport serialization, you can not introduce new serialization in 7.17.x
+                        // since it will break serialization for mixed clusters with 7.17.x and earlier versions of 8.x. This hack uses the
+                        // metadata field to serialize the excludeDeletedDocs flag.
+                        Map<String, Object> metadata = new HashMap<>(termsBuilder.getMetadata());
+                        metadata.put("exclude_deleted_docs", true);
```

Comment:
Should we use something like `_exclude_deleted_docs` because this could conceivably overlap with a user-supplied metadata key?

Reply:
https://github.com/elastic/elasticsearch/pull/109494/commits/468476e924ea03241beb94289f5a53d69b226164